In [1]:
import pandas as pd
import plotly.express as px

In [2]:
df = pd.read_csv(r'C:\Users\Gayathri\OneDrive\Desktop\cardeckho\processed_final_data.csv')

In [3]:
df.sample()

manufacturer  model_year body_type transmission_type fuel_type  total_kms  \
391       Maruti        2016     Sedan         Automatic    Petrol      30000   

     total_owners     price         insurance_type  mileage(kmpl)  ...  \
391             1  725000.0  Third Party insurance          19.12  ...   

     torque(nm)  seats  number_of_cylinders  valves_per_cylinder  \
391       130.0    5.0                    4                    4   

     super_charger turbo_charger drive_type steering_type cargo_volume  \
391             No            No        FWD         Power        510.0   

      location  
391  Bangalore  

[1 rows x 21 columns]

##### **Pearson Correlation**

In [4]:
numerical_variables = ['total_kms','price' ,'mileage(kmpl)' ,'engine(CC)', 'cargo_volume']

corr_df = df[numerical_variables]

corr_data = corr_df.corr()
corr_data

total_kms     price  mileage(kmpl)  engine(CC)  cargo_volume
total_kms       1.000000 -0.074043      -0.013922    0.091595      0.003116
price          -0.074043  1.000000      -0.249676    0.530541      0.006896
mileage(kmpl)  -0.013922 -0.249676       1.000000   -0.521500     -0.030719
engine(CC)      0.091595  0.530541      -0.521500    1.000000      0.015596
cargo_volume    0.003116  0.006896      -0.030719    0.015596      1.000000

In [5]:
fig = px.imshow(corr_data, text_auto = True)
fig.show()

- ##### Engine(CC) seems to have a postive correlation of 0.53 with the target variable (Price)